### ISSUE TITLE
3.0.0.pre9 installation fails on Windows

### ISSUE THREAD

**Issue by aharpervc** [scoutapp/scout_apm_ruby#91]
My environment is mri 2.3.3 (x86, mingw32) on Windows 10. I'm trying to upgrade from 1.6.8 to 3.0.0.pre9, but the gem install is failing.

```sh
$ gem install scout_apm -v '3.0.0.pre9'
Temporarily enhancing PATH to include DevKit...
Building native extensions.  This could take a while...
ERROR:  Error installing scout_apm:
        ERROR: Failed to build gem native extension.

(...snip...)

generating allocations-i386-mingw32.def
compiling allocations.c
allocations.c:13:45: fatal error: sys/resource.h: No such file or directory
compilation terminated.
make: *** [allocations.o] Error 1

make failed, exit code 2

(...snip...)
```

mkmf.log doesn't have anything helpful, but it doesn't matter. The problem is `sys/resource.h` (and probably also `sys/time.h`) headers & some linux/posix specific stuff.

Now, this is just for development. It is not super critical that scoutapm be collecting allocations for my dev environment, so I think the simplest thing here is to just `#ifdef _WIN32` some stuff.

I have done that here, in my fork, which solves the problem on my machine:
https://github.com/scoutapp/scout_apm_ruby/compare/v3.0.0.pre9...aharpervc:fix-windows-installation?expand=1

I'm happy to open a pr & work through your testing/ci system, but this is based off of the `v3.0.0.pre9` tag which I can't seem to base a pr against. Is there a branch that tracks that work I can base it on?

**Comment by cschneid** [comment]
Great bug report & bit of code. Thank you!

Our 3.0.0.x versions are being developed on the `scoutprof` branch if you'd like to make a PR.

**Comment by aharpervc** [comment]
@cschneid fyi https://github.com/scoutapp/scout_apm_ruby/pull/92

**Comment by dlanderson** [comment]
Fixed in #92
